rust/feature-flags/src/cohort_cache.rs

```diff
@@ -0,0 +1,221 @@
+use crate::api::FlagError;
+use crate::cohort_models::Cohort;
+use crate::flag_matching::{PostgresReader, TeamId};
+use moka::future::Cache;
+use std::time::Duration;
+
+/// CohortCacheManager manages the in-memory cache of cohorts using `moka` for caching.
+///
+/// Features:
+/// - **TTL**: Each cache entry expires after 5 minutes.
+/// - **Size-based eviction**: The cache evicts least recently used entries when the maximum capacity is reached.
+///
+/// ```text
+/// CohortCacheManager {
+///     postgres_reader: PostgresReader,
+///     per_team_cohorts: Cache<TeamId, Vec<Cohort>> {
+///         // Example:
+///         2: [
+///             Cohort { id: 1, name: "Power Users", filters: {...} },
+///             Cohort { id: 2, name: "Churned", filters: {...} }
+///         ],
+///         5: [
+///             Cohort { id: 3, name: "Beta Users", filters: {...} }
+///         ]
+///     }
+/// }
+/// ```
+///
+#[derive(Clone)]
+pub struct CohortCacheManager {
+    postgres_reader: PostgresReader,
+    per_team_cohort_cache: Cache<TeamId, Vec<Cohort>>,
+}
```

Comment:
NB: I call this struct `CohortCacheManager` because the cache itself is actually the `per_team_cohort_cache`, I just liked having the ability to have this class manage the DB connection too (which I need to seed/update the cache entries).